packages/mgt-element/src/providers/Providers.ts

```diff
@@ -112,6 +116,55 @@ export class Providers {
     return this._me;
   }
 
+  /**
+   * Gets the cache ID, creates one if it does not exist
+   *
+   * @static
+   * @memberof Providers
+   */
+  public static async getCacheId(force?: boolean) {
+    if (this._cacheId && !force) {
+      return this._cacheId;
+    }
+    if (Providers.globalProvider && Providers.globalProvider.state == ProviderState.SignedIn) {
+      if (!this._cacheId || force) {
+        this._cacheId = null;
+        const client = this.client;
+        if (client) {
+          try {
+            const response: User = await client.api('me').get();
```

Comment:
this will get called multiple times unnecessarily - I'm seeing 4 calls with just few components on the page. Save the promise in a field and await the saved promise to ensure only one call to graph is made.